app/controllers/EventReportController.scala

```diff
@@ -124,20 +112,21 @@ class EventReportController @Inject()(
     }
   }
 
-  private def get(block: (String, String, String) => Future[Result])
-                 (implicit hc: HeaderCarrier, request: Request[AnyContent]): Future[Result] = {
+  private def withAuthAndOverviewParameters(block: (String, String, String, String) => Future[Result])
+                                           (implicit hc: HeaderCarrier, request: Request[AnyContent]): Future[Result] = {
 
     authorised(Enrolment("HMRC-PODS-ORG") or Enrolment("HMRC-PODSPP-ORG")).retrieve(Retrievals.externalId) {
       case Some(_) =>
         (
           request.headers.get("pstr"),
+          request.headers.get("reportType"),
           request.headers.get("startDate"),
           request.headers.get("endDate")
         ) match {
-          case (Some(pstr), Some(startDate), Some(endDate)) =>
-            block(pstr, startDate, endDate)
+          case (Some(pstr), Some(reportType), Some(startDate), Some(endDate)) =>
+            block(pstr, reportType, startDate, endDate)
           case _ =>
-            Future.failed(new BadRequestException("Bad Request with missing PSTR/ Start Date/ End Date"))
```

Comment:
the error message you are reporting is a bit generic , I would suggest the following changes  
        val pstrMissing =   if(request.headers.get("pstr")).isDefined "" else "pstr missing"
       val reportTypeMissing =   if(request.headers.get("pstr")).isDefined "" else "reportType missing"
      val startDateMissing =   if(request.headers.get("pstr")).isDefined "" else "startDate missing"
      val endDateMissing =   if(request.headers.get("pstr")).isDefined "" else "endDate missing"

Future.failed(new BadRequestException(s"Bad Request with missing $pstrMissing  $reportTypeMissing  $startDateMissing  $endDateMissing "))

Reply:
Sorted. Code pushed.